### PR TITLE
igt-gpu-tools: update to 1.24, fix musl

### LINKED ISSUE #11374: Package Request: igt-gpu-tools for musl
[These][1] are Intel's official GPU monitoring tools, and it'd be good to have them packaged. Thanks to the efforts of @Johnnynator, [there's a musl patch][2] (against HEAD). Additionally, ``-Dbuild_tests=no -Dbuild_runner=no`` needs to be passed to Meson in that case.

[1]: https://gitlab.freedesktop.org/drm/igt-gpu-tools
[2]: https://gist.github.com/Johnnynator/e0aa4bbcfe662adf373ab00825d8bd57

### PR DESCRIPTION
**Pull request by st3r4g**
Closes #11374

### CODE REVIEW

File: srcpkgs/igt-gpu-tools/template [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="https://gitlab.freedesktop.org/drm/igt-gpu-tools"
 distfiles="${XORG_SITE}/app/${pkgname}-${version}.tar.xz"
-checksum=4d4b086c513bace5c23d0889de3f42ac3ebd3d968c64dedae6e28e006a499ad0
+checksum=57357c72feeafc923c9cfd2d1234bd80e120fc7cc6099eac81158ec351a821bf
 
 lib32disabled=yes
-archs="i686 x86_64"
+archs="i686 i686-musl x86_64 x86_64-musl"
```
Comment (Johnnynator) [inline srcpkgs/igt-gpu-tools/template]: You could just use wildcards
```suggestion
archs="i686* x86_64*"
```

File: srcpkgs/igt-gpu-tools/patches/musl.patch [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+ #pragma GCC diagnostic ignored "-Wpointer-arith"
+diff --git lib/signal_compat.h lib/signal_compat.h
+new file mode 100644
+index 00000000..a37c5820
+--- /dev/null
++++ b/lib/signal_compat.h
+@@ -0,0 +1,8 @@
++#define SIGEV_SIGNAL 0
++#define SIGEV_NONE 1
++#define SIGEV_THREAD 2
```
Comment (Johnnynator) [inline srcpkgs/igt-gpu-tools/patches/musl.patch]: Remove this block

**Comment by st3r4g** [comment]
Some warnings due to `musl.patch`:
```
[14/366] Compiling C object 'benchmarks/cb1d2fd@@gem_syslatency@exe/gem_syslatency.c.o'.
../benchmarks/gem_syslatency.c: In function 'sys_wait':
../benchmarks/gem_syslatency.c:180:29: warning: assignment to 'void (*)(union sigval)' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  180 |  sev.sigev_notify_thread_id = gettid();
      |                             ^
[52/366] Compiling C object 'lib/76b5a35@@igt-igt_aux_c@sta/igt_aux.c.o'.
../lib/igt_aux.c: In function 'igt_sigiter_start':
../lib/igt_aux.c:187:30: warning: assignment to 'void (*)(union sigval)' from 'pid_t' {aka 'int'} makes pointer from integer without a cast [-Wint-conversion]
  187 |   sev.sigev_notify_thread_id = __igt_sigiter.tid;
      |                              ^
../lib/igt_aux.c: In function 'igt_start_siglatency':
../lib/igt_aux.c:1615:29: warning: assignment to 'void (*)(union sigval)' from 'int' makes pointer from integer without a cast [-Wint-conversion]
 1615 |  sev.sigev_notify_thread_id = gettid();
      |                             ^
```
There's still the problem that the build sometimes hangs, tried adding `disable_parallel_build=yes` but no success. I could reach the end when manually chrooting though.

**Comment by st3r4g** [comment]
The point in which the build hangs is random, and is likely caused by one of the default flags passed to meson by `build_style=meson`, since the build succeeds with plain `do_build() { meson build }`

File: srcpkgs/igt-gpu-tools/template [hunk 3]
```diff
@@ -1,20 +1,30 @@
...
-archs="i686 x86_64"
+archs="i686* x86_64*"
+
+case "$XBPS_TARGET_MACHINE" in
+	*-musl) configure_args+=" -Dbuild_tests=disabled -Dbuild_runner=disabled -Dbuild_docs=disabled"
+esac
+
+# Some default flag makes the build hang in a random point
+do_configure() {
+	meson build
```
Comment (Johnnynator) [inline srcpkgs/igt-gpu-tools/template]: NO. figure out what causes the issue and revert it it configure_args. Currently you are just making sure that the build does not fail, but everything get's installed in wrong locations, you don't pass the musl configure_args, etc...

Comment (Johnnynator) [reply srcpkgs/igt-gpu-tools/template]: Add `-Db_lto=false` to configure args and add a comment that it takes unusually long on glibc with lto on.

Comment (Johnnynator) [reply srcpkgs/igt-gpu-tools/template]: Okay, you found it already out :D

Comment (st3r4g) [reply srcpkgs/igt-gpu-tools/template]: I was just doing that

**Comment by st3r4g** [comment]
Fully fixed musl now, tests cdon't have to be disabled.

**Comment by st3r4g** [comment]
Tested some of the tools

File: srcpkgs/igt-gpu-tools/template [hunk 4]
```diff
@@ -1,20 +1,30 @@
...
+checksum=57357c72feeafc923c9cfd2d1234bd80e120fc7cc6099eac81158ec351a821bf
 
 lib32disabled=yes
-archs="i686 x86_64"
+archs="i686* x86_64*"
+
+build_options="tests"
+desc_option_tests="Build tests"
+
+if [ ! ${build_option_tests} ]; then
```
Comment (xtraeme) [inline srcpkgs/igt-gpu-tools/template]: This should check for XBPS_CHECK_PKGS instead (i.e check target).

Comment (xtraeme) [reply srcpkgs/igt-gpu-tools/template]: get rid of the build option and use:

```
if [ $XBPS_CHECK_PKGS ]]; then
 ...
fi
```

Comment (st3r4g) [reply srcpkgs/igt-gpu-tools/template]: done

**Pull request merged**